File: codes/ruby/chapter01/answer_A04.rb
# 入力
n = gets.to_i

# 上の桁から順番に「2進法に変換した値」を求める
(9).downto(0) do |x|
  wari = 1 << x # 2のx乗
  print (n / wari) % 2 # 割り算の結果に応じて0または1の出力
end
puts # 最後に改行する

File: codes/go/chapter01/answer_A04.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)

	// 上の桁から順番に「2進法に変換した値」を求める
	for x := 9; x >= 0; x-- {
		wari := 1 << x            // 2のx乗
		fmt.Print((n / wari) % 2) // 割り算の結果に応じて0または1を出力
	}
	fmt.Println()
}
